### DISCUSSION TITLE
Is it me or the PWA plugin isn't really generating the service worker?

### DISCUSSION THREAD

**Discussion by JeremieLitzler** [vuepress-theme-hope/vuepress-theme-hope#3095] [Q&A]
Hello,

In my Netlify build logs, I see that the service worker is generated:
![image](https://user-images.githubusercontent.com/2410556/235516105-3f98d8f9-49e1-46c6-b09e-96f673bb7666.png)

But when I browser my site () which has a valid manifest (), there is no service worker being displayed in the DevTools > Applications > Service worker tab:

![image](https://user-images.githubusercontent.com/2410556/235516027-e1cacad5-9bfd-45a3-be5d-34b09ce6d50f.png)

I see the same result on https://theme-hope.vuejs.press/

Anyone has the same issue?

**Comment by Mister-Hope** [comment]
The official theme no longer use pwa, while my blog still is and I think it's working, so you'd better check again yourself, and provide a minimal reproduction if possible

**Reply by JeremieLitzler** [reply]
Hi @Mister-Hope 

I'm not sure to understand.

You say *The official theme no longer use pwa* but [the docs](https://theme-hope.vuejs.press/guide/advanced/pwa.html) don't say it is unavailable... Or am I missing something?

I use the minimum configuration (see my [theme.ts](https://github.com/JeremieLitzler/journal/blob/main/docs/.vuepress/theme.ts)).

I read carefully the changelogs on each new version... Have I missing something?

The difference between your blog and mine is found in the manifest declaration (I have a seperate file, you configure it theme.ts). But I know this difference has been so for a long time now.

**Reply by Mister-Hope** [reply]
The theme docs no longer use pwa, that's what I mean, so you are expected to find nothing on the application page.

**Reply by JeremieLitzler** [reply]
Hi  @Mister-Hope 

Out of curiosity, what motivated the drop of usage?

Also, I notice that a `service-worker.js` file is still available. Does it mean I can register the service worker myself?

**Reply by JeremieLitzler** [reply]
Hi @Mister-Hope 

I'd love to know how is your blog still registering a service worker using your theme... Can you explain please?

Is it because you detail the manifest in `theme.ts`?

Thanks a million.

**Reply by Mister-Hope** [reply]
Emm, some update here, I am busy and I do not have time to check your project, so you need to at least provide a minimal reproduction to show the case that pwa is not working.
